go/common/cbor/cbor.go

```diff
@@ -27,14 +27,23 @@ var (
 		TagsMd:        cbor.TagsForbidden,
 	}
 
+	// decOptions are decoding options for UNTRUSTED inputs (used by default).
 	decOptions = cbor.DecOptions{
 		DupMapKey:   cbor.DupMapKeyEnforcedAPF,
 		IndefLength: cbor.IndefLengthForbidden,
 		TagsMd:      cbor.TagsForbidden,
 	}
 
-	encMode cbor.EncMode
-	decMode cbor.DecMode
+	// decOptionsTrusted are decoding options for TRUSTED inputs. They are only used when explicitly
+	// requested by using the UnmarshalTrusted method.
+	decOptionsTrusted = cbor.DecOptions{
+		MaxArrayElements: 134217728, // Maximum allowed.
+		MaxMapPairs:      134217728, // Maximum allowed.
```

Comment:
what are the defaults for these on the untrusted `decOptions`?

Reply:
[The default is 131072.](https://github.com/fxamacker/cbor#decoding-options) Maybe we should consider making this explicit and/or even changing it (e.g., lowering it) for untrusted settings?

Reply:
eh, maybe a comment if anything

Reply:
@kostko hello. I have an issue with calling [StateToGenesis beacon api](https://github.com/oasisprotocol/oasis-core/blob/9b98fdcad2c64935df98a98a9aba55de9cbe0f06/go/beacon/api/grpc.go#L219): `rpc error: code = Internal desc = grpc: failed to unmarshal the received message cbor: exceeded max number of key-value pairs 131072 for CBOR map` at block `7188323`. How I can explicitly call UnmarshalTrusted method at this point?